modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/StatisticsEndpoint.java

```diff
@@ -0,0 +1,383 @@
+        .map(ProviderQuery::new)
+        .flatMap(q ->
+          statisticsService
+            .getProvider(q.providerId)
+            .map(Stream::of).orElseGet(Stream::empty)
+            .peek(p -> checkAccess(q.resourceId, p.getResourceType()))
+            .map(p -> timeSeriesToJson(
+              p.getId(),
+              statisticsService.getTimeSeriesData(p, q.resourceId, q.from, q.to, q.dataResolution,
+                ZoneId.systemDefault()))))
+        .forEach(result::add);
+    } catch (IllegalArgumentException e) {
+      return RestUtil.R.badRequest(e.getMessage());
+    }
+    return Response.ok(result.toJSONString()).build();
+  }
+
+  private JSONObject timeSeriesToJson(String providerId, TimeSeries timeSeriesData) {
+    final JSONObject result = new JSONObject();
+    result.put("providerId", providerId);
+    result.put("providerType", "timeSeries");
+    result.put("labels", timeSeriesData.getLabels());
+    result.put("values", timeSeriesData.getValues());
+    if (timeSeriesData.getTotal().isPresent()) {
+      result.put("total", timeSeriesData.getTotal().getAsDouble());
+    }
+    return result;
+  }
+
+  private void checkAccess(final String resourceId, final ResourceType resourceType) {
+    try {
+      switch (resourceType) {
+        case EPISODE:
+          checkMediapackageAccess(resourceId);
+          break;
+        case SERIES:
+          checkSeriesAccess(resourceId);
+          break;
+        case ORGANIZATION:
+          checkOrganizationAccess(resourceId);
+          break;
+        // Thanks CheckStyle, very sensible
+        default:
+          break;
+      }
+    } catch (UnauthorizedException | SearchIndexException e) {
+      chuck(e);
+    }
+  }
+
+  private void checkMediapackageAccess(final String mpId) throws UnauthorizedException, SearchIndexException {
+    final Opt<Event> event = indexService.getEvent(mpId, searchIndex);
+    if (event.isNone()) {
+      // IndexService checks permissions and returns None if user is unauthorized
+      throw new UnauthorizedException(securityService.getUser(), "read");
+    }
+  }
+
+  private void checkSeriesAccess(final String seriesId) throws UnauthorizedException, SearchIndexException {
+    final Opt<Series> series = indexService.getSeries(seriesId, searchIndex);
+    if (series.isNone()) {
+      // IndexService checks permissions and returns None if user is unauthorized
+      throw new UnauthorizedException(securityService.getUser(), "read");
+    }
+  }
+
+  private void checkOrganizationAccess(final String orgId) throws UnauthorizedException {
+    final User currentUser = securityService.getUser();
+    final Organization currentOrg = securityService.getOrganization();
+    final String currentOrgAdminRole = currentOrg.getAdminRole();
+    final String currentOrgId = currentOrg.getId();
+
+    boolean authorized = currentUser.hasRole(GLOBAL_ADMIN_ROLE)
+      || (currentUser.hasRole(currentOrgAdminRole) && currentOrgId.equals(orgId));
```

Comment:
There is no way to view organization statistics if you are not an admin or an org admin? What is the `ROLE_UI_STATISTICS_ORGANIZATION_VIEW` for?

I think we can change the access check to:
1. if the use is admin, provide access
2. if the user is org admin, provide access
3. if the user have role `ROLE_UI_STATISTICS_ORGANIZATION_VIEW` and the `orgID` is the ID oth the users organization, provide access
4. otherwise throw an exception

Are you agree?

Reply:
No, a user that has not access to all events of an organization should not see that statistics.

Or why should that be the case?

Reply:
The user you describing may not have the `ROLE_UI_STATISTICS_ORGANIZATION_VIEW` if he/she shouldn't see the org statistics. But in case you want to show your stats, you can add the role for the user and you are not limited to org admins.

Reply:
Fixed. Users with `ROLE_UI_STATISTICS_ORGANIZATION_VIEW` can now view the org stats if they belong to the organization.